MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1435,6 +1468,28 @@ private async Task<bool> RefreshLocalTrailers(IHasTrailers item, MetadataRefresh
             return itemsChanged;
         }
 
+        private async Task<bool> RefreshExtras(BaseItem item, MetadataRefreshOptions options, List<FileSystemMetadata> fileSystemChildren, CancellationToken cancellationToken)
+        {
+            var newExtras = LoadExtras(fileSystemChildren, options.DirectoryService).Concat(LoadThemeVideos(fileSystemChildren, options.DirectoryService)).Concat(LoadThemeSongs(fileSystemChildren, options.DirectoryService));
+
+            var newExtraIds = newExtras.Select(i => i.Id).ToArray();
+
+            var extrasChanged = !item.ExtraIds.SequenceEqual(newExtraIds);
```

Comment:
why don't you return early if `extrasChanged` is false?

Reply:
Changed that now and still works. It was mainly the same behavior as in `RefreshThemeVideos` that I used as the source for this routine.